autogluon/task/image_classification/classifier.py

```diff
@@ -112,37 +114,83 @@ def predict(self, X, input_size=224, plot=True):
         >>> test_data = task.Dataset('~/data/test', train=False)
         >>> class_index, class_probability = classifier.predict('example.jpg')
         """
-        # model inference
+
         input_size = self.model.input_size if hasattr(self.model, 'input_size') else input_size
-        resize = int(math.ceil(input_size / 0.875))
-        transform_fn = Compose([
-                Resize(resize),
-                CenterCrop(input_size),
-                ToTensor(),
-                transforms.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225])
-            ])
-        def predict_img(img):
+        resize = int(math.ceil(input_size / resize_ratio))
+
+        transform_size = transforms.Compose([
+            transforms.Resize(resize),
+            transforms.CenterCrop(input_size),
+            transforms.ToTensor(),
+            transforms.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225])
+        ])
+
+        def predict_img(img, ensemble=False):
             proba = self.predict_proba(img)
-            ind = mx.nd.argmax(proba, axis=1).astype('int')
-            idx = mx.nd.stack(mx.nd.arange(proba.shape[0], ctx=proba.context), ind.astype('float32'))
-            probai = mx.nd.gather_nd(proba, idx)
-            return ind, probai, proba
+            if ensemble:
+                return proba
+            else:
+                ind = mx.nd.argmax(proba, axis=1).astype('int')
+                idx = mx.nd.stack(mx.nd.arange(proba.shape[0], ctx=proba.context), ind.astype('float32'))
+                probai = mx.nd.gather_nd(proba, idx)
+                return ind, probai, proba
+
+        def avg_prediction(different_dataset, threshold=0.001):
+            result = {}
+            inds, probas, probals_all = [], [], []
+            for i in range(len(different_dataset)):
+                items = len(different_dataset[0])
+                for j in range(items):
+                    result.setdefault(j, []).append(different_dataset[i][j])
+            for c in result.keys():
+                proba_all = sum([*result[c]]) / len(different_dataset)
+                proba_all = mx.nd.array(np.where(proba_all.asnumpy() >= threshold, proba_all.asnumpy(), 0))
+                ind = mx.nd.argmax(proba_all, axis=1).astype('int')
+                idx = mx.nd.stack(mx.nd.arange(proba_all.shape[0], ctx=proba_all.context), ind.astype('float32'))
+                proba = mx.nd.gather_nd(proba_all, idx)
+                inds.append(ind.asscalar())
+                probas.append(proba.asnumpy())
+                probals_all.append(proba_all.asnumpy().flatten())
+            return inds, probas, probals_all
+
+        def predict_imgs(X):
```

Comment:
can you lowercase the argument (PEP8)  maybe with a more semantic name? Also docstring

Reply:
> can you lowercase the argument (PEP8) maybe with a more semantic name? Also docstring

Thanks for your reminder i will add it immediately.